common/queue/src/test/java/org/thingsboard/server/queue/discovery/ZkDiscoveryServiceTest.java

```diff
@@ -0,0 +1,173 @@
+import static org.mockito.Mockito.when;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ZkDiscoveryServiceTest {
+
+    @Mock
+    private TbServiceInfoProvider serviceInfoProvider;
+
+    @Mock
+    private PartitionService partitionService;
+
+    @Mock
+    private CuratorFramework client;
+
+    @Mock
+    private PathChildrenCache cache;
+
+    private ScheduledExecutorService zkExecutorService;
+
+    @Mock
+    private CuratorFramework curatorFramework;
+
+    private ZkDiscoveryService zkDiscoveryService;
+
+    @Before
+    public void setup() {
+        zkDiscoveryService = Mockito.spy(new ZkDiscoveryService(serviceInfoProvider, partitionService));
+        zkExecutorService = Executors.newSingleThreadScheduledExecutor(ThingsBoardThreadFactory.forName("zk-discovery"));
+        when(client.getState()).thenReturn(CuratorFrameworkState.STARTED);
+        ReflectionTestUtils.setField(zkDiscoveryService, "stopped", false);
+        ReflectionTestUtils.setField(zkDiscoveryService, "client", client);
+        ReflectionTestUtils.setField(zkDiscoveryService, "cache", cache);
+        ReflectionTestUtils.setField(zkDiscoveryService, "nodePath", "/thingsboard/nodes/0000000010");
+        ReflectionTestUtils.setField(zkDiscoveryService, "zkExecutorService", zkExecutorService);
+        ReflectionTestUtils.setField(zkDiscoveryService, "recalculateDelay", 1000L);
+        ReflectionTestUtils.setField(zkDiscoveryService, "zkDir", "/thingsboard");
+    }
+
+    @Test
+    public void restartNodeTest() throws Exception {
+        var currentInfo = TransportProtos.ServiceInfo.newBuilder().setServiceId("currentId").build();
+        var currentData = new ChildData("/thingsboard/nodes/0000000010", null, currentInfo.toByteArray());
+        var childInfo = TransportProtos.ServiceInfo.newBuilder().setServiceId("childId").build();
+        var childData = new ChildData("/thingsboard/nodes/0000000020", null, childInfo.toByteArray());
+
+        when(serviceInfoProvider.getServiceInfo()).thenReturn(currentInfo);
+        List<ChildData> dataList = new ArrayList<>();
+        dataList.add(currentData);
+        when(cache.getCurrentData()).thenReturn(dataList);
+
+        startNode(childData);
+
+        verify(partitionService, times(1)).recalculatePartitions(eq(currentInfo), eq(List.of(childInfo)));
+
+        reset(partitionService);
+
+        //Restart in timeAssert.assertTrue(zkDiscoveryService.delayedTasks.isEmpty());
+        stopNode(childData);
+
+        assertEquals(1, zkDiscoveryService.delayedTasks.size());
+
+        verify(partitionService, never()).recalculatePartitions(eq(currentInfo), any());
+
+        startNode(childData);
+
+        verify(partitionService, never()).recalculatePartitions(eq(currentInfo), any());
+
+        Thread.sleep(2000);
+
+        verify(partitionService, never()).recalculatePartitions(eq(currentInfo), any());
+
+        assertTrue(zkDiscoveryService.delayedTasks.isEmpty());
+
+        //Restart not in time
```

Comment:
Please, split this into 3 or more tests, and remove empty lines. I can not open the whole test on Full HD laptop :)